src/recommendpolicy/recommendPolicy.go

```diff
@@ -143,22 +160,46 @@ func RecommendPolicyMain() {
 			DeployNsName = append(DeployNsName, *deploy)
 		}
 
-		for _, ns := range nsNotFilter {
+		for _, ns := range NsNotFilter {
 			if d.Namespace != ns {
-				generateHardenPolicy(d.Name, d.Namespace, d.Spec.Template.Labels)
+				policy = generateHardenPolicy(d.Name, d.Namespace, d.Spec.Template.Labels)
 			}
 		}
 	}
+	for _, pol := range policy {
+		policies = append(policies, pol)
+	}
+	for _, r := range replicaSets.Items {
+		for _, ns := range NsNotFilter {
+			if r.Namespace != ns {
+				policy = generateHardenPolicy(r.Name, r.Namespace, r.Spec.Template.Labels)
+			}
+		}
+	}
+	for _, pol := range policy {
+		policies = append(policies, pol)
+	}
+	for _, r := range statefulSets.Items {
+		for _, ns := range NsNotFilter {
+			if r.Namespace != ns {
+				policy = generateHardenPolicy(r.Name, r.Namespace, r.Spec.Template.Labels)
+			}
+		}
+	}
+	for _, pol := range policy {
+		policies = append(policies, pol)
```

Comment:
We are not using this `policies` anywhere after it is returned right? Do we need this policies for any operation?

Reply:
We are using [GetHardenPolicy()](https://github.com/accuknox/discovery-engine/pull/681/files#diff-507e0011278270a66cc4717d5f666421567b9ca156b62c4c01670f52e6867a44R225) to get the policies for testing, RecommendPolicyMain() will directly not return it therefore using this func.